tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -1038,10 +1038,158 @@ def test_with_dotnetcore(self, runtime, code_uri, mode, architecture="x86_64"):
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_with_dotnetcore_in_container_mount_with_write_explicit(
+        self, runtime, code_uri, mode, architecture="x86_64"
+    ):
+        overrides = {
+            "Runtime": runtime,
+            "CodeUri": code_uri,
+            "Handler": "HelloWorld::HelloWorld.Function::FunctionHandler",
+            "Architectures": architecture,
+        }
+
+        if runtime == "provided.al2":
+            self.template_path = self.template_path.replace("template.yaml", "template_build_method_dotnet_7.yaml")
+
+        # test with explicit mount_with_write flag
+        cmdlist = self.get_command_list(use_container=True, parameter_overrides=overrides, mount_with_write=True)
+        # env vars needed for testing unless set by dotnet images on public.ecr.aws
+        cmdlist += ["--container-env-var", "DOTNET_CLI_HOME=/tmp/dotnet"]
+        cmdlist += ["--container-env-var", "XDG_DATA_HOME=/tmp/xdg"]
+
+        LOG.info("Running Command: {}".format(cmdlist))
+        LOG.info("Running with SAM_BUILD_MODE={}".format(mode))
+
+        newenv = os.environ.copy()
+        if mode:
+            newenv["SAM_BUILD_MODE"] = mode
+
+        run_command(cmdlist, cwd=self.working_dir, env=newenv)
+
+        self._verify_built_artifact(
+            self.default_build_dir,
+            self.FUNCTION_LOGICAL_ID,
+            self.EXPECTED_FILES_PROJECT_MANIFEST
+            if runtime != "provided.al2"
+            else self.EXPECTED_FILES_PROJECT_MANIFEST_PROVIDED,
+        )
+
+        self._verify_resource_property(
+            str(self.built_template),
+            "OtherRelativePathResource",
+            "BodyS3Location",
+            os.path.relpath(
+                os.path.normpath(os.path.join(str(self.test_data_path), "SomeRelativePath")),
+                str(self.default_build_dir),
+            ),
+        )
+
+        self._verify_resource_property(
+            str(self.built_template),
+            "GlueResource",
+            "Command.ScriptLocation",
+            os.path.relpath(
+                os.path.normpath(os.path.join(str(self.test_data_path), "SomeRelativePath")),
+                str(self.default_build_dir),
+            ),
+        )
+
+        expected = "{'message': 'Hello World'}"
+        if not SKIP_DOCKER_TESTS:
+            self._verify_invoke_built_function(
+                self.built_template, self.FUNCTION_LOGICAL_ID, self._make_parameter_override_arg(overrides), expected
+            )
+        self.verify_docker_container_cleanedup(runtime)
+
+    @parameterized.expand(
+        [
+            ("dotnetcore3.1", "Dotnetcore3.1", None),
+            ("dotnet6", "Dotnet6", None),
+            ("dotnetcore3.1", "Dotnetcore3.1", "debug"),
+            ("dotnet6", "Dotnet6", "debug"),
+            ("provided.al2", "Dotnet7", None),
+        ]
+    )
+    @pytest.mark.flaky(reruns=3)
+    def test_with_dotnetcore_in_container_mount_with_write_interactive(
```

Comment:
Same feedback regarding skipping Docker test decorator and `if` statement check at the end of this test